canine/orchestrator.py

```diff
@@ -463,119 +527,126 @@ def make_output_DF(self, batch_id, outputs, cpu_time, prev_acct, localizer = Non
                     pandas_write_hdf5_buffered(df, buf = w, key = "results")
         return df
 
-    def submit_batch_job(self, entrypoint_path, compute_env, extra_sbatch_args = {}) -> int:
+    def submit_batch_job(self, entrypoint_path, compute_env, extra_sbatch_args = {}, job_spec = None) -> int:
+        if job_spec is None:
+            job_spec = self.job_spec
+
         # this job was avoided
-        if len(self.job_spec) == 0:
+        if len(job_spec) == 0:
             return -2
 
         batch_id = self.backend.sbatch(
             entrypoint_path,
+            "H", # submit in "held" state to allow for noop'd jobs to be cancelled
             **{
                 'requeue': True,
                 'job_name': self.name,
-                'array': "0-{}".format(len(self.job_spec)-1),
+                'array': "0-{}".format(len(job_spec)-1),
                 'output': "{}/%a/stdout".format(compute_env['CANINE_JOBS']),
                 'error': "{}/%a/stderr".format(compute_env['CANINE_JOBS']),
                 **self.resources,
                 **Orchestrator.stringify(extra_sbatch_args)
             }
         )
 
+        # cancel noop'd jobs
+        for k, v in job_spec.items():
+            if v is None:
+                self.backend.scancel(batch_id + "_" + k)
+
+        # release the batch
+        # TODO: should scontrol be added to the backend class? AFAIK this is the
```

Comment:
It wouldn't _hurt_ but if this is the only time we use scontrol, it's definitely not important